File: SmartQuitIoT-Mobile/smartquitiot/lib/models/request/appointment_request.dart
// lib/models/request/appointment_request.dart
class AppointmentRequest {
  final int coachId;
  final int slotId;
  final String date; // yyyy-MM-dd

  AppointmentRequest({
    required this.coachId,
    required this.slotId,
    required this.date,
  });

  Map<String, dynamic> toJson() => {
    'coachId': coachId,
    'slotId': slotId,
    'date': date,
  };

  factory AppointmentRequest.fromJson(Map<String, dynamic> j) => AppointmentRequest(
    coachId: j['coachId'] as int,
    slotId: j['slotId'] as int,
    date: j['date'] as String,
  );
}

File: SmartQuitIoT-BE/src/main/java/com/smartquit/smartquitiot/dto/request/AppointmentRequest.java
package com.smartquit.smartquitiot.dto.request;

import jakarta.validation.constraints.FutureOrPresent;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class AppointmentRequest {
  @NotNull(message = "Coach ID is required")
  int coachId;

  @NotNull(message = "slot ID is required")
  int slotId;

  @NotNull(message = "Date is required")
  @FutureOrPresent(message = "Date must be today or in the future")
  LocalDate date;
}
